types/src/validator_set.rs

```diff
@@ -71,33 +66,55 @@ impl ValidatorSetResource {
         self.last_reconfiguration_time
     }
 
-    pub fn validator_set(&self) -> &ValidatorSet {
-        &self.validator_set
+    pub fn validator_set(&self) -> ValidatorSet {
+        ValidatorSet::new(self.scheme, self.validators.clone())
     }
 }
 
+#[derive(Clone, Copy, Debug, Deserialize, Eq, PartialEq, Serialize)]
+#[cfg_attr(any(test, feature = "fuzzing"), derive(Arbitrary))]
+#[repr(u8)]
+pub enum ConsensusScheme {
+    Ed25519 = 0,
+}
```

Comment:
While this overall works with one approach, the problem is if we actually do have other variants, we cannot easily move between them in Rust.